File: euler_021.py
from time import time

def sum_of_proper_factors(n):
    return sum([d for d in range(1, n // 2 + 1) if n%d == 0])

start = time()

queue = list(range(1,10000))

amicable = set()

while queue:
    a = queue[0]
    b = sum_of_proper_factors(a)
    c = sum_of_proper_factors(b)
    if a != b and a == c:
        amicable.add(a)
    queue.remove(a)

print(sum(amicable))

stop = time()
print("Time: " + str(stop-start))

File: euler_021.js
const start = Date.now();

let sum = 0;
let sumDict = {};

const sumOfProperFactors = (num) => {
    let max = num;
    let factors = [1];
    for (let i = 2;i < max;i++) {
        if (num % i === 0) {
            factors.push(i);
            let s = num / i;
            if (s !== num) {
                factors.push(s);
                max = s;
            }
        }
    }
    return factors.reduce((t,n) => t + n,0);
}

for (let i = 2; i < 10000; i++) {
    sumDict[i] = sumOfProperFactors(i);
}

for (let i = 2; i < 10000; i++) {
    if (sumDict[sumDict[i]] === i && sumDict[i] !== i) {
        sum += i;
    }
}

const end = Date.now();

console.log("Execute time: ", end - start, " Sum of amicable numbers: ", sum);